src/assignments/tech-jobs-console.rst

```diff
@@ -0,0 +1,492 @@
+   actionChoices.Add("search", "Search");
+   actionChoices.Add("list", "List");
+
+If you recall how the program worked when you ran it, the first menu
+that you chose had two options, *Search* and *List*, which seem to
+correspond to the entries in ``actionChoices``. This is, in fact, the
+case. This is the data that is used to generate the first menu we see
+when running the program.
+
+The second usage of ``GetUserSelection`` is a few lines below:
+
+.. sourcecode:: c#
+   :lineno-start: 35
+
+   string columnChoice = GetUserSelection("List", columnChoices);
+
+This references ``columnChoices``, which is declared at the top of
+``Main`` and has a similar structure to ``actionChoices`` (they’re the
+same data type and are used in calls to the same method, so this
+shouldn’t be surprising). Most of the entries in ``columnChoices``
+correspond to columns in the jobs data set, but there’s one additional
+entry with key/value pair ``"all"``/``"All"``. These entries will help
+us present to the user the options for searching our data, which will
+correspond to searching within a given column, or searching all columns
+at once.
+
+The keys in ``actionChoices`` and ``columnChoices`` represent the
+*internal* strings we’ll use to refer to these options (e.g. when representing
+the user’s menu choice, or querying data). The values in the dictionary represent the
+*external* way that these are represented to the user.
+
+.. TODO: replace this microsoft link with an internal ref to the dowhile section
+
+Within ``GetUserSelection`` itself, most of the code is within a
+``do-while loop``. Recall that a `do-while
+loop <https://msdn.microsoft.com/en-us/library/370s1zax.aspx>`__
+is similar to a ``while`` loop, but the conditional check is at the
+*end* of the loop’s code block. This has the net consequence that the
+loop’s code block *always runs at least once*. At the end of the block’s
+execution, we check a condition to determine if we should run the block
+again. This nicely mimics the behavior of simple menu-driven
+applications.
+
+Within this loop, menu options are printed to the screen, and user input
+is collected. If the input is valid, it returns the choice as a string
+to the caller. This string corresponds to the chosen key (from
+``choices``, which will be either ``actionChoices`` or
+``columnChoices``) of the item the user selected. If invalid, it
+re-prompts the user.
+
+The local variable ``choiceKeys`` is used to easily enumerate the
+``choices`` dictionary. In other words, it gives us a simple way to
+provide an ordering to ``choices``, which doesn’t have an ordering of
+its own.
+
+The ``JobData`` Class
+---------------------
+
+The ``JobData`` class is responsible for importing the data from the CSV
+file and parsing it into a C#-friendly format; that is, into
+``Dictionary`` and ``List`` form. Look toward the bottom of the class
+and you will see a method named ``LoadData``, which does just what it
```

Comment:
let's keep it to encourage them to search for it themselves

Reply:
Then I would change the region that you point them to because ``LoadData`` is pretty much in the middle. I would either say, "Look toward the middle of the class" or "If you scroll down the page, you will see a method named ``LoadData`` ..."